google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java

```diff
@@ -621,7 +637,8 @@ Builder setCustomerEncryption(CustomerEncryption customerEncryption) {
 
     @Override
     Builder setKmsKeyName(String kmsKeyName) {
-      this.kmsKeyName = kmsKeyName;
+      this.kmsKeyName =
+          kmsKeyName != null && kmsKeyName.contains("cryptoKeyVersions") ? "" : kmsKeyName;
```

Comment:
This change should be in the HttpStorageRpc request being sent instead.

Reply:
done